oxd-gen-client/src/test/java/io/swagger/client/api/RemoveSiteTest.java

```diff
@@ -0,0 +1,56 @@
+package io.swagger.client.api;
+
+import io.swagger.client.ApiResponse;
+import io.swagger.client.model.RegisterSiteResponseData;
+import io.swagger.client.model.RemoveSiteParams;
+import io.swagger.client.model.UpdateSiteResponse;
+import io.swagger.client.model.UpdateSiteResponseData;
+import org.apache.commons.lang.StringUtils;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class RemoveSiteTest {
+
+    @Test
+    @Parameters({"opHost", "redirectUrl"})
+    public void testRemoveSite(String opHost, String redirectUrl) throws Exception {
+        final DevelopersApi api = Tester.api();
+        RegisterSiteResponseData response = RegisterSiteTest.registerSite(api, opHost, redirectUrl);
+        String oxdId = response.getOxdId();
+        RemoveSiteParams params = new RemoveSiteParams();
+        params.setOxdId(oxdId);
+        ApiResponse<UpdateSiteResponse> apiResponse =
+                api.removeSiteWithHttpInfo(Tester.getAuthorization(), params);
+        assertTrue(apiResponse.getStatusCode() == 200);
+        assertNotNull(apiResponse);
+        assertNotNull(apiResponse.getData());
+        /*
+        Swagger hub doesn't generate RemoveSiteResponse for some reason, so we have to
+        use UpdateSiteResponse.
+         */
```

Comment:
minor but please remove this comment, we are all clear about it.

Reply:
I was hoping it might help a new entrant when he/she goes through this.

Reply:
I can recommend to read `'Clean code' by Robert C. Martin`. The basic rules are:
- Always try to explain yourself in code.
- Don't be redundant.
- Don't add obvious noise.
- Don't use closing brace comments.
- Don't comment out code. Just remove.
- Use as explanation of intent.
- Use as clarification of code.
- Use as warning of consequences.

Reply:
Thanks for suggestion :). I think I am using it for -> Use as clarification of code. I understand it is clear to me and even more clear to you why we have it here. However, IMHO,  at first glance, to anyone else not aware of the issue,  it may look like a bug (Why would any one use an updateYyyy() method to remove an entity instead of a removeYyyy() method ).

Reply:
wait-wait, not update method, method is `removeSite`, the issue is that response is called `UpdateSiteResponse` but for anyone using swagger it should be clear that swagger is optimizing and re-use classes if they are identical. Also remember that real clients will never see it since they will just generate code. If we wish to explain things like this then it is better to put it in docs.

Reply:
All right :). I will change it. However, what you are seeing as optimization I am seeing a bug in swagger's code generation mechanism. Also, by "anyone" mentioned in my post, I mean the developer who is going to maintain this test suite, and not the client, who would just generate the code itself.